sdk/log/batch.go

```diff
@@ -43,19 +64,39 @@ func NewBatchingProcessor(exporter Exporter, opts ...BatchingOption) *BatchingPr
 		exporter = defaultNoopExporter
 	}
 	cfg := newBatchingConfig(opts)
-	return &BatchingProcessor{
+	b := &BatchingProcessor{
 		exporter: exporter,
 
 		maxQueueSize:       cfg.maxQSize.Value,
 		exportInterval:     cfg.expInterval.Value,
 		exportTimeout:      cfg.expTimeout.Value,
 		exportMaxBatchSize: cfg.expMaxBatchSize.Value,
+
+		flush: make(chan exportRequest),
+		stop:  make(chan exportRequest, 1),
+		done:  make(chan struct{}),
+		queue: make([]Record, 0, cfg.maxQSize.Value),
+		batch: make([]Record, 0, cfg.expMaxBatchSize.Value),
 	}
+
+	go b.run()
+	return b
 }
 
 // OnEmit batches provided log record.
 func (b *BatchingProcessor) OnEmit(ctx context.Context, r Record) error {
-	// TODO (#5063): Implement.
+	if b.isShutdown.Load() {
+		return nil
+	}
+
+	defer b.mu.Unlock()
+	b.mu.Lock()
+
+	if len(b.queue) == b.maxQueueSize {
+		// Queue is full.
```

Comment:
Should we trigger a flush when the queue is full? The span batching processor is dropping therefore I decided to do the same. Maybe it should be configurable?

I think it would be better to track it is a separate issue.